src/mono/mono/sgen/sgen-memory-governor.c

```diff
@@ -210,6 +210,10 @@ sgen_memgov_minor_collection_end (const char *reason, gboolean is_overflow)
 
 		sgen_add_log_entry (log_entry);
 	}
+
+	sgen_gc_info.heap_size_bytes = sgen_major_collector.get_num_major_sections () * sgen_major_collector.section_size + sgen_los_memory_usage_total;
+	sgen_gc_info.fragmented_bytes = sgen_gc_info.heap_size_bytes - sgen_los_memory_usage - (last_used_slots_size + sgen_total_allocated_major - total_allocated_major_end);
```

Comment:
note that sweep can run concurrently with minor collections, so computations here might be broken

Reply:
Oh, I didn't realize that. Ok; this won't work either. Hmm...

Reply:
I expected it to work here because it is nw using last_used_slots_size, which is the value after the last sweep completed. But apparently there is some other problem with doing this here.